Cron/EventsTopic.php

```diff
@@ -0,0 +1,190 @@
+<?php
+
+namespace Klaviyo\Reclaim\Cron;
+
+use Klaviyo\Reclaim\Helper\Logger;
+use Klaviyo\Reclaim\Model\SyncsFactory;
+use Klaviyo\Reclaim\Model\Quote\QuoteIdMask;
+use Klaviyo\Reclaim\Model\Resourcemodel\Events\CollectionFactory;
+
+use Magento\Catalog\Model\CategoryFactory;
+
+class EventsTopic
+{
+    /**
+     * Klaviyo Logger
+     * @var Logger
+     */
+    protected $_klaviyoLogger;
+
+    /**
+     * Klaviyo QuoteIdMask ResourceModel
+     * @var QuoteIdMask
+     */
+    protected $_quoteIdMaskResource;
+
+    /**
+     * Klaviyo Events Collection Factory
+     * @var CollectionFactory
+     */
+    protected $_eventsCollectionFactory;
+
+    /**
+     * Klaviyo Sync Factory
+     * @var SyncsFactory
+     */
+    protected $_klSyncFactory;
+
+    /**
+     * Magento Category Factory
+     * @var CategoryFactory
+     */
+    protected $_categoryFactory;
+
+    /**
+     * @param Logger $klaviyoLogger
+     * @param CollectionFactory $eventsCollectionFactory
+     * @param SyncsFactory $klSyncFactory
+     * @param QuoteIdMask $quoteIdMaskResource
+     * @param CategoryFactory $categoryFactory
+     */
+    public function __construct(
+        Logger $klaviyoLogger,
+        CollectionFactory $eventsCollectionFactory,
+        SyncsFactory $klSyncFactory,
+        QuoteIdMask $quoteIdMaskResource,
+        CategoryFactory $categoryFactory
+    )
+    {
+        $this->_klaviyoLogger = $klaviyoLogger;
+        $this->_eventsCollectionFactory = $eventsCollectionFactory;
+        $this->_klSyncFactory = $klSyncFactory;
+        $this->_quoteIdMaskResource = $quoteIdMaskResource;
+        $this->_categoryFactory = $categoryFactory;
+    }
+
+    /**
+     * Gets 500 rows from the kl_events table and move to the kl_sync table
+     */
+    public function moveRowsToSync()
+    {
+        // New Events to be moved to kl_sync table and update status of these to Moved, limit 500
+        $eventsCollection = $this->_eventsCollectionFactory->create();
+        $eventsData = $eventsCollection->getRowsForSync('NEW')
+            ->addFieldToSelect(['id','event','payload','user_properties'])
+            ->getData();
+
+        if (empty($eventsData)){
+            return;
+        }
+
+        $idsMoved = [];
+
+        // Capture all events that have been moved and add data to Sync table
+        foreach ($eventsData as $event){
+            if ($event['topic'] == 'Added To Cart') {
```

Comment:
I think this is a bug, kl_events table doesn't have column named 'topic'
```suggestion
            if ($event['event'] == 'Added To Cart') {
```